analyzers/ruby/ruby.go

```diff
@@ -127,47 +127,92 @@ func (a *Analyzer) IsBuilt() (bool, error) {
 }
 
 func (a *Analyzer) Analyze() (graph.Deps, error) {
-	lockfilePath := filepath.Join(a.Module.Dir, "Gemfile.lock")
-	if a.Options.LockfilePath != "" {
-		lockfilePath = a.Options.LockfilePath
+	shouldFallback := true
```

Comment:
Is `shouldFallback` still needed?

Reply:
yes. We don't know if `list-lockfile` is explicitly provided or implicitly chosen.